### PR TITLE
feat(cli): cross-provider run management — get/runs/--json/--detach, manage-run skill, orchestrator delivery

### PR DESCRIPTION
**Pull request by Wirasm**
## Summary

- **Problem:** The native `manage_run` tool only works on Claude + Pi; Codex/OpenCode/Copilot have no in-process tool hook. The CLI was the natural cross-provider surface but had four gaps (no single-run detail, active-only `status`, non-universal `--json`, blocking `run`), and nothing told the chat agent it could use the CLI.
- **Why it matters:** Closes run management for **all five providers** via the one surface every bash-capable agent already has — no served MCP, no per-provider in-process tool.
- **What changed:** Added `workflow get` / `workflow runs`; threaded `--json` into `approve`/`reject`/`abandon`/`resume`; added `--detach`; shipped a focused `manage-run` skill (bundled + installed + CI-guarded); and added a "Managing Workflow Runs" section to the orchestrator system prompt so chat on any provider can drive the CLI.
- **What did NOT change (scope boundary):** The native `manage_run` tool stays (complementary). No served MCP. No new core DB queries (reused `getWorkflowRun`/`listDashboardRuns`). No direct-chat `skills:` plumbing (workflow-node-only today). No `archon`-on-PATH detection (inherits the existing `archon` skill convention).

### Before

```
Agent (Codex/OpenCode/Copilot) in a project, via bash:
  $ archon workflow list          → workflow DEFINITIONS (not runs)
  $ archon workflow status        → only running/paused (no history)
  $ archon workflow <id>          → ✗ no such command
  $ archon workflow approve <id>  → human text only (hard to parse)
  $ archon workflow run X "msg"   → BLOCKS until the workflow finishes
  Chat agent: no idea it can manage runs at all.
```

### After

```
Agent in a project, via bash (cwd → auto-scoped):
  $ archon workflow runs --json          → recent runs (ALL statuses), this project
  $ archon workflow get <id> --json      → one run: status, error, [per-node]
  $ archon workflow approve <id> --json  → { "ok": true, "action": "approve", "resumable": true }
  $ archon workflow run X "msg" --detach → returns immediately; child runs it
  [*] Chat on any provider: system prompt now teaches these verbs directly.
```

### After

```
  cli.ts ──▶ commands/workflow.ts ──▶ @archon/core (getWorkflowRun, listDashboardRuns)  [reused]
                     │
                     ├─[+] workflowGetCommand / workflowRunsCommand
                     ├─[+] --json on approve/reject/abandon/resume (record/validate, no inline exec)
                     └─[+] --detach → Bun.spawn detached re-invoke (~/.archon/logs/)

  bundled-skill.ts ─[+]─ BUNDLED_MANAGE_RUN_SKILL_FILES ──▶ skill.ts install ──▶ .claude/skills/manage-run/
                                                            scripts/check-bundled-skill.ts (allowlist guard)

  orchestrator-agent.ts ──▶ prompt-builder.ts ─[+]─ buildRunManagementSection()  (all providers, direct chat)
```

**Connection inventory:**

| From | To | Status | Notes |
|------|----|--------|-------|
| `commands/workflow.ts` | `@archon/core` db queries | unchanged | reused `getWorkflowRun`, `listDashboardRuns` |
| `cli.ts` | `commands/workflow.ts` | **modified** | new subcommands + flags; `--json` implies log suppression |
| `bundled-skill.ts` | `.claude/skills/manage-run/` | **new** | embedded text imports + new map |
| `prompt-builder.ts` | orchestrator system prompt | **modified** | appended run-management section |

## Label Snapshot

- Risk: `risk: low`
- Size: `size: M`
- Scope: `cli`, `core:orchestrator`, `skills`, `docs`, `tests`
- Module: `cli:workflow`, `core:orchestrator`

## Change Metadata

- Change type: `feature`
- Primary scope: `multi` (cli + core)

## Linked Issue

- Closes # — none (PRP-driven; plans archived under `.claude/PRPs/`)

## Validation Evidence (required)

```bash
bun run validate    # → exit 0
```

- `workflow.test.ts`: 120 pass (was 113); `prompt-builder.test.ts`: 15 pass (was 11); full CLI suite + all packages green; full `bun run validate` exit 0 (type-check, lint, format, check:bundled/-skill/-schema, all tests).
- Manual (dev): `get`/`runs`/`--json` produce clean parseable stdout; `skill install` writes both skills (23 files); CI guard fails on an unbundled skill file; `Bun.spawn`+`unref()` child verified to survive parent `process.exit()`.

## Security Impact (required)

- New permissions/capabilities? **No**
- New external network calls? **No**
- Secrets/tokens handling changed? **No**
- File system access scope changed? **Yes (minor)** — `--detach` writes child stdout/stderr to a per-conversation log file under `~/.archon/logs/` (graceful fallback to discard if unopenable). This is *additive* and exists specifically to avoid silently swallowing a detached child's bootstrap errors.

## Compatibility / Migration

- Backward compatible? **Yes** (all additions; existing human output unchanged byte-for-byte)
- Config/env changes? **No**
- Database migration needed? **No**

## Human Verification (required)

- Verified scenarios: `get`/`runs` human + `--json` round-trip via `JSON.parse`; `--json` now clean on stdout (Pino default destination is stdout → `--json` implies log suppression); `skill install` into a probe dir; CI-guard negative test; detached-child survival.
- Edge cases checked: unregistered cwd → global list + explicit note; invalid `--status` → fail-fast; write-command `--json` error path returns `{ ok: false }` instead of throwing.
- What was not verified: `--detach` in a **compiled binary** build (implemented symmetrically via `BUNDLED_IS_BINARY`, dev path empirically verified; binary path unit-tested for argv only).

## Side Effects / Blast Radius (required)

- Affected subsystems: `@archon/cli` (workflow commands + entry), bundled-skill pipeline, `core/orchestrator` system prompt.
- Potential unintended effects: the orchestrator system prompt gains a section on every chat — additive guidance only; routing rules unchanged.
- Guardrails: full `bun run validate` in CI; `check:bundled-skill` allowlist guard.

## Rollback Plan (required)

- Fast rollback: revert the two commits (`feat(cli): …`, `feat(orchestrator): …`) — self-contained, no migrations, no state.
- Feature flags: none needed.
- Observable failure symptoms: `archon workflow get/runs` erroring; `--json` output non-parseable; chat agent failing to find `archon` (PATH — same convention as the existing `archon` skill).

## Risks and Mitigations

- Risk: `--detach` re-invoke fragile across dev/binary.
  - Mitigation: mode-detected via `BUNDLED_IS_BINARY`; argv unit-tested; dev path + child-survival empirically verified; child output logged (not discarded) so failures aren't silent.
- Risk: CI-guard wrongly requiring local-only skill dirs (repo has 13).
  - Mitigation: explicit `BUNDLED_SKILLS = ['archon','manage-run']` allowlist; negative test confirms it fails only on a missing *bundled* file.
- Risk: orchestrator prompt assumes `archon` on PATH.
  - Mitigation: inherits the existing `archon` skill's convention (no new assumption); section is additive guidance.

## Summary by CodeRabbit

* **New Features**
  * Start workflows in the background with --detach (per-run log files) and new workflow get / workflow runs commands; workflow control commands support machine-readable --json acknowledgements that record decisions without inline execution.

* **Documentation**
  * Expanded CLI and docs to cover detach, --json output, runs/get/status semantics, scopeFallback behavior, and install instructions for the bundled manage-run skill.

* **Tests**
  * Added CLI tests for runs listing/status/get, JSON behavior, detached runs, and control commands.

### CODE REVIEW

File: packages/cli/src/commands/workflow.ts [hunk 1]
```diff
@@ -983,42 +1124,193 @@ export async function workflowStatusCommand(json?: boolean, verbose?: boolean):
...
+    throw new Error(`Failed to get workflow run: ${err.message}`);
+  }
+
+  if (!run) {
+    if (json) {
+      console.log(JSON.stringify({ error: 'not_found', runId }, null, 2));
+    } else {
+      console.log(`Workflow run not found: ${runId}`);
+    }
+    return;
```
[the remark this answers is not in the document]
Comment (Wirasm) [reply packages/cli/src/commands/workflow.ts]: Fixed in `795c4985`. `get --json` now never throws and emits the consistent `{ ok: false, runId, error }` envelope on both DB error and not-found (was a throw + `{ error: "not_found" }`). Applied the same never-throw treatment to `runs --json` (invalid `--status` + DB error), and added tests for both. Resolved.

File: packages/core/src/orchestrator/prompt-builder.test.ts [hunk 2]
```diff
@@ -134,4 +135,37 @@ describe('buildOrchestratorSystemAppend', () => {
...
+    expect(section).toContain('## Managing Workflow Runs');
+    for (const verb of [
+      'archon workflow runs',
+      'archon workflow get',
+      'archon workflow status',
+      'archon workflow run',
+      'archon workflow approve',
+      'archon workflow reject',
+      'archon workflow abandon',
+    ]) {
```
[the remark this answers is not in the document]
Comment (Wirasm) [reply packages/core/src/orchestrator/prompt-builder.test.ts:163]: Done in `795c4985` — added `archon workflow resume` to the verb assertions (and to the prompt section itself, per the sibling comment). Resolved.

File: packages/core/src/orchestrator/prompt-builder.ts [hunk 3]
```diff
@@ -212,10 +212,37 @@ ${formatProjectSection(scopedCodebase)}
...
+You can inspect and control this project's workflow runs directly via the \`archon\` CLI (bash) — you do NOT need to invoke a workflow for run management. Add \`--json\` to any command for a single clean, machine-readable line.
+
+- \`archon workflow runs [--json]\` — recent runs of ALL statuses for this project
+- \`archon workflow get <run-id> [--json]\` — one run's status/error (add \`--verbose\` for per-node detail)
+- \`archon workflow status [--json]\` — active runs only (running/paused)
+- \`archon workflow run <workflow> "<message>" --detach\` — start a run in the background (returns immediately)
+- \`archon workflow approve <run-id> [--json]\` / \`archon workflow reject <run-id> [reason] [--json]\` — resolve a paused approval gate
+- \`archon workflow abandon <run-id> [--json]\` — cancel a non-terminal run
+
+When the user asks what's running, whether a run passed/failed, or to approve / reject / cancel a run, use these commands directly instead of invoking a workflow. The \`manage-run\` skill has the full reference if it is loaded.`;
```
[the remark this answers is not in the document]
Comment (Wirasm) [reply packages/core/src/orchestrator/prompt-builder.ts]: Good catch — valid and important. `archon workflow …` is gated behind a git-repo cwd in `cli.ts`, so from `~/.archon/workspaces/` it would error `Not in a git repository`. Fixed in `795c4985`: the section now opens with a "run from within the project's git repo (any subdirectory works — resolves to the repo root, which also scopes `runs` to this project)" preface, including the explicit failure note. Resolved.

File: packages/core/src/orchestrator/prompt-builder.ts [hunk 4]
```diff
@@ -212,10 +212,37 @@ ${formatProjectSection(scopedCodebase)}
...
+  return `## Managing Workflow Runs
+
+You can inspect and control this project's workflow runs directly via the \`archon\` CLI (bash) — you do NOT need to invoke a workflow for run management. Add \`--json\` to any command for a single clean, machine-readable line.
+
+- \`archon workflow runs [--json]\` — recent runs of ALL statuses for this project
+- \`archon workflow get <run-id> [--json]\` — one run's status/error (add \`--verbose\` for per-node detail)
+- \`archon workflow status [--json]\` — active runs only (running/paused)
+- \`archon workflow run <workflow> "<message>" --detach\` — start a run in the background (returns immediately)
+- \`archon workflow approve <run-id> [--json]\` / \`archon workflow reject <run-id> [reason] [--json]\` — resolve a paused approval gate
+- \`archon workflow abandon <run-id> [--json]\` — cancel a non-terminal run
```
[the remark this answers is not in the document]
Comment (Wirasm) [reply packages/core/src/orchestrator/prompt-builder.ts:241]: Added in `795c4985` — `archon workflow resume <run-id>` is now in the run-control list (noting it re-runs failed/paused runs skipping completed nodes; run as a background task, `--json` validates only). Resolved.

**Comment by Wirasm** [comment]
## PR Review Summary — multi-agent

Seven specialized agents reviewed this PR (code quality, docs impact, tests, comments, silent failures, type design, simplification). Findings below are deduped and cross-corroborated; the top correctness/doc items were re-verified by hand against the PR branch. Finding IDs use `C/I/S` prefixes (not `#`) to avoid cross-issue autolinks.

**Verdict: NEEDS FIXES** — no merge blockers (no data-loss / security / happy-path crash), but several should-fix correctness and doc-accuracy items, three of which were independently flagged by 2–3 agents.

---

### Critical Issues (0 found)
None. The No-Autonomous-Lifecycle-Mutation rule is respected, the never-throw `--json` contract holds on the happy path, and there is no data-loss or security regression.

---

### Important Issues (8 found)

| ID | Issue | Location | Agents |
|----|-------|----------|--------|
| I1 | **`--json` stdout corruption on DB hiccup.** Pino defaults to **stdout** (`pino({ level })`, no destination — confirmed in `paths/src/logger.ts`), and `--json` only lowers the level to `warn` (not silence). The `getLog().warn(...)` at the codebase-lookup catch is reachable *inside* the `--json` path, so a slow/unreachable DB prints a log line ahead of the JSON and breaks `JSON.parse` — exactly on the `archon workflow runs --json` polling command the manage-run skill depends on. **Verified.** Recommended systemic fix: route the CLI Pino logger to **stderr** (fixes I6 too). | `workflow.ts:1237` | silent-failure, code-reviewer |
| I2 | **`WorkflowRunOptions.json` JSDoc is wrong** — says "only honored with --detach", but the field is honored by `approve`/`reject`/`abandon`/`resume` independent of `--detach`. Misleads any caller reading the public interface. **3-agent consensus.** | `workflow.ts:85` | code-reviewer, comment-analyzer, type-design |
| I3 | **CLAUDE.md describes reversed behavior.** Bullet says the run-management section reaches "**any** provider" / "**all five providers**", but `orchestrator-agent.ts:1057` gates it to `!scopedCaps.nativeTools` — it reaches only Codex/OpenCode/Copilot; Claude/Pi get the native `manage_run` tool instead. Since CLAUDE.md is read by AI agents, this actively misleads. **Verified.** | `CLAUDE.md:500` | docs-impact |
| I4 | **Public CLI reference missing all new surface.** `docs-web/.../reference/cli.md` documents none of `workflow get`, `workflow runs`, `--detach`, or `--json` on the four write commands; `workflow status` description is stale; `skill install` now installs two skills (`archon` + `manage-run`). Users consulting the canonical reference find nothing. | `packages/docs-web/src/content/docs/reference/cli.md` | docs-impact |
| I5 | **`workflow get` not-found exits 0** (both human and `--json`). Breaks `&&` chaining / CI exit-code checks for the command marketed as the polling primitive. Return a non-zero code (or propagate one) on not-found. | `workflow.ts:1162-1168` | silent-failure |
| I6 | **Detached child inherits `--json`.** `spawnDetachedWorkflowRun` filters only `--detach` from the child argv, so the child runs with `--json` → suppresses its own `info` logs in the `~/.archon/logs/` file, partially defeating the log's stated "don't fail silently" purpose. Filter `--json` too (or fix Pino→stderr per I1). | `workflow.ts` (`spawnDetachedWorkflowRun`) | code-reviewer |
| I7 | **Empty `catch {}` swallows `listWorkflowEvents` errors** in the `--verbose` path of both `workflowGetCommand` and `workflowStatusCommand`. A failed event query renders identically to "run has no nodes" — no log, no user note. Add a `warn` log + a `(node events unavailable)` note. | `workflow.ts ~1175` & `~1122` | silent-failure |
| I8 | **Comment contradictions.** (a) Three-way contradiction on resume detection: `workflow.ts:582` says "executor no longer performs implicit resume detection" while the docstring + inline comment at `:1365` say "the executor's findResumableRun" picks it up. (b) `workflowApproveCommand` docstring claims "transitions to failed for auto-resume" unconditionally, but `--json` mode suppresses auto-resume. | `workflow.ts:582 / 1365`, approve docstring | comment-analyzer |

---

### Suggestions (lower priority)

| ID | Suggestion | Location | Agents |
|----|------------|----------|--------|
| S1 | FD leak: `openSync` fd not closed if `Bun.spawn` throws. Wrap spawn in `try/finally` + `closeSync`. (Benign on immediate exit, but a real correctness gap on the binary execPath-wrong path.) | `workflow.ts ~141-150` | code-reviewer, silent-failure |
| S2 | `workflow runs --json` doesn't surface the scope-fallback. When cwd is unregistered (or lookup fails) it silently returns the **global** list with no `scopeFallback` field — an agent can't distinguish "empty, project-scoped" from "global fallback". Add `scopeFallback: true` to the JSON. | `workflow.ts ~1259` | silent-failure, type-design |
| S3 | Stale comment `// Skill Files (21 total)` → now 23 (21 archon + 2 manage-run imports). **Verified.** | `bundled-skill.ts:12` | comment-analyzer |
| S4 | `Number(values.limit)` yields `NaN` on `--limit abc` and flows into `listDashboardRuns`. Validate as a positive integer and fail fast. | `cli.ts:500` | type-design |
| S5 | DRY (Rule of Three satisfied — 4 sites): extract a `jsonError(runId, action, error)` helper for the identical write-command catch blocks. ok-true payloads stay explicit (they differ). | `workflow.ts` write cmds | code-simplifier |
| S6 | Micro-simplifications: redundant `codebase?.id ?? undefined` → `codebase?.id`; extract the ternary inside `JSON.stringify` in `workflowGetCommand`; conditional-spread → pass `undefined` directly (verify `listDashboardRuns` sig first). | `workflow.ts:1183,1241,1244` | code-simplifier |
| S7 | Test gaps: (crit 7) `--detach` **binary-mode** argv (`BUNDLED_IS_BINARY=true`) is untested — the only path where a wrong-argv silent spawn could ship; (crit 6) orchestrator `nativeTools` gating (CLI-pointer vs native-tool) is never exercised — mock always returns `nativeTools` undefined; plus `reject --json` `cancelled:false` case. | `workflow.test.ts`, `orchestrator-agent.test.ts` | pr-test-analyzer |
| S8 | The six `--json` envelopes are an agent-facing contract expressed entirely as anonymous inline literals (expression 4/10, enforcement 5/10). Consider named types / Zod for the write-command success+error shapes so envelope drift is caught at compile time. Low priority for a CLI, but real given the consumer is an agent. | `workflow.ts` (envelopes) | type-design |

---

### Documentation Issues
- `CLAUDE.md:500` — reversed-behavior bullet (I3)
- `packages/docs-web/.../reference/cli.md` — missing `get`/`runs`/`--detach`/`--json`; stale `status` + `skill install` (I4)
- `packages/docs-web/.../getting-started/overview.md` — CLI table missing new commands; stale `status` description; manual `cp -r` skill-install example copies only `archon`
- `packages/docs-web/.../guides/skills.md` — `manage-run` missing from the bundled-skills table
- `packages/docs-web/.../contributing/cli-internals.md:21` — `workflow.ts` comment omits `get`/`runs`
- `bundled-skill.ts:12` — stale `(21 total)` (S3)
- Skill docs (`manage-run/SKILL.md`, `references/commands.md`, `archon/references/cli-commands.md`) were verified **accurate and complete** — no changes needed.

---

### Strengths
- No-Autonomous-Lifecycle-Mutation rule respected — every state transition is an explicit user action.
- The never-throw `--json` contract is correctly upheld across all six commands in the final state (a mid-PR inconsistency in `workflow get`'s error shape was caught and fixed in a fixup commit).
- `--status` validation reuses `workflowRunStatusSchema.safeParse()` — no parallel ad-hoc validation.
- Codebase-scoping never *silently* wrong-scopes on the human path (explicit "showing all runs" note).
- `check-bundled-skill.ts` allowlist guard is correctly scoped and documents its substring-match limitation; negative-test confirms it fails only on a missing bundled file.
- Strong docstrings on `spawnDetachedWorkflowRun` (documents the rejected alternative) and the `--json` log-suppression rationale in `cli.ts`.
- Thorough behavioral test coverage for `workflow get` / `workflow runs` (not-found, invalid status fail-fast, JSON shapes, `--all`, scope fallback).

---

### Recommended Actions
1. **I1** (and I6) — route the CLI Pino logger to **stderr**; this is the one fix that closes both the `--json` corruption and the detached-child log suppression at the source.
2. **I3 + I4** — correct the `CLAUDE.md` orchestrator bullet and backfill the `docs-web` CLI reference (highest doc-accuracy impact; I3 misleads agents).
3. **I2, I5, I7, I8** — fix the JSDoc, the `workflow get` exit code, the empty `catch {}` events path, and the contradictory comments.
4. Consider **S1–S4** (fd leak, scope-fallback field, stale count, limit validation) and the **S7** test gaps before merge; **S5/S6/S8** are polish.

_Re-run targeted aspects (`code`, `errors`, `docs`) after fixes._
